src/control/cmd/daos/pool.go

```diff
@@ -276,6 +279,96 @@ func (cmd *poolQueryCmd) Execute(_ []string) error {
 	return nil
 }
 
+type poolQueryTargetsCmd struct {
+	poolBaseCmd
+
+	Rank    uint32 `long:"rank" required:"1" description:"Engine rank of the targets to be queried"`
+	Targets string `long:"target-idx" description:"Comma-separated list of target idx(s) to be queried"`
+}
+
+func convertDaosSpaceInfo(in *C.struct_daos_space, mt C.uint) *mgmtpb.StorageTargetUsage {
+	if in == nil {
+		return nil
+	}
+
+	return &mgmtpb.StorageTargetUsage{
+		Total:     uint64(in.s_total[mt]),
+		Free:      uint64(in.s_free[mt]),
+		MediaType: mgmtpb.StorageMediaType(mt),
+	}
+}
+
+// For using the pretty printer that dmg uses for this target info.
+//
+// Does the same thing as ds_mgmt_drpc_pool_query_target() will do, to stuff the target info
+// into a protobuf message, then use the automatic conversion from proto to control.
+
+func convertPoolTargetInfo(ptinfo *C.daos_target_info_t) (*control.PoolQueryTargetInfo, error) {
+	pqtp := new(mgmtpb.PoolQueryTargetInfo)
+
+	pqtp.Type = mgmtpb.PoolQueryTargetInfo_TargetType(ptinfo.ta_type)
+	pqtp.State = mgmtpb.PoolQueryTargetInfo_TargetState(ptinfo.ta_state)
+	pqtp.Space = []*mgmtpb.StorageTargetUsage{
+		convertDaosSpaceInfo(&ptinfo.ta_space, C.DAOS_MEDIA_SCM),
+		convertDaosSpaceInfo(&ptinfo.ta_space, C.DAOS_MEDIA_NVME),
+	}
+
+	// Manual conversion mgmtpb -> control; convert.Types not working for some reason (type, state)
+	pqti := new(control.PoolQueryTargetInfo)
+	pqti.Type = control.PoolQueryTargetType(pqtp.Type)
+	pqti.State = control.PoolQueryTargetState(pqtp.State)
+	pqti.Space = []*control.StorageTargetUsage{
+		{pqtp.Space[uint(C.DAOS_MEDIA_SCM)].Total, pqtp.Space[uint(C.DAOS_MEDIA_SCM)].Free, C.DAOS_MEDIA_SCM},
+		{pqtp.Space[uint(C.DAOS_MEDIA_NVME)].Total, pqtp.Space[uint(C.DAOS_MEDIA_NVME)].Free, C.DAOS_MEDIA_NVME},
+	}
```

Comment:
If we're doing this either way, why not populate it directly instead of going through the intermediate step with the protobuf.

Reply:
I think perhaps in following the model of the existing daos/dmg pool query I left it this way, with the intention to use convert.Types. But I simply could not get that to completely work. Perhaps this code can be streamlined now to avoid the conversion altogether (manual or automatic).

Reply:
removed protobuf usage here in latest commits